### PR TITLE
fix: properly update dependency map

### LINKED ISSUE #9168: [Bug]: Api form body doesn't update when its dependent on an entity which initially did not exist even after entity gets created
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

[View loom video](
[LOOM DEMO](https://www.loom.com/share/4256e89f65f2483eaf12e14488ce797c)

### Steps To Reproduce

[View loom video](
[LOOM DEMO](https://www.loom.com/share/4256e89f65f2483eaf12e14488ce797c)

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by ohansFavour**
## Description

This PR in conjunction with  [8981](https://github.com/appsmithorg/appsmith/pull/8981), fixes #9168 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
Jest 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>
No changes to code coverage between the base branch and the head branch</details>

### CODE REVIEW

File: app/client/src/workers/DataTreeEvaluator.ts [hunk 1]
```diff
@@ -518,6 +504,23 @@ export default class DataTreeEvaluator {
         });
       }
     }
+
```
Comment (ohansFavour) [inline app/client/src/workers/DataTreeEvaluator.ts:507]: I have re-ordered the position of this code block. The dependencies generated from the dynamicBindingPathList were getting cleared.

**Review by ApekshaBhosale — APPROVED** [review]

**Pull request merged**